### DISCUSSION TITLE
Exists any way to implement Prisma in AWS Lambda layers?

### DISCUSSION THREAD

**Discussion by bcarusodev** [prisma/prisma#17936] [Q&A] [answered]
Hi everyone, i'm creating a backend repo with Prisma & TypeScript and Serverless Framework (with `serverless-bundle` script) on AWS. 

When i import Prisma client to lambdas, compilation ends in ~40MB lambdas, it's not a problem for now because the limit is on 50MB but if in a future i'm going to implement more features maybe will have a problem.

So, i think we can put Prisma on a lambda layer and share across all lambdas that interacts with my DB, it is possible with Serverless Framework? 

Thanks! :)

**Comment by nurul3101** [comment, marked as answer]
Hello @silvergraphs 👋 

We have a feature request for providing AWS Lambda layers here:
- https://github.com/prisma/prisma/issues/12506

Based on the conversation in the above feature requests, some users were able to host prisma engines in the Lambda layers.
Also, we are working to ensure that Prisma works seamlessly in serverless environments and one of its initiatives is to reduce the bundle size. We recently shipped updates in Prisma [4.8.0](https://github.com/prisma/prisma/releases/4.8.0) that reduced engine size by 50%, we are continuing our work to further enhance serverless support for Prisma.

**Reply by nurul3101** [reply]
Also, here is a blog that might be helpful to you: https://medium.com/ama-tech-blog/deploying-prisma-with-aws-lambda-layers-in-cdk-10608f5598c5

**Comment by raymondKelly** [comment]
Just my 2 cents on this area. In all the blogs and posts I see on here I see the entire client being pushed into a layer. I disagree with this approach. You should allow the bundler to do tree shaking and only include what code is needed for that particular Lambda. What I would like to put in the Layer is the large binary file instead.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
